Bottom: 6f155a817b55a1d4380c1dc69dd45314d1499f68
Top:    f60ef76093d0c5e4d7e11e61cf21a733e79cc0e0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-10 02:47:42 +0000

Refresh of remove-my-memory-allocation-ro

---

diff --git a/src/create.c b/src/create.c
index 18928d3..538ea0d 100644
--- a/src/create.c
+++ b/src/create.c
@@ -1006,8 +1006,8 @@ CreateNewAttribute (AttributeListTypePtr list, char *name, char *value)
       list->Max += 10;
       list->List = realloc (list->List, list->Max * sizeof (AttributeType));
     }
-  list->List[list->Number].name = strdup (name);
-  list->List[list->Number].value = strdup (value);
+  list->List[list->Number].name = STRDUP (name);
+  list->List[list->Number].value = STRDUP (value);
   list->Number++;
   return &list->List[list->Number - 1];
 }
diff --git a/src/file.c b/src/file.c
index bfe0de0..ecfd71d 100644
--- a/src/file.c
+++ b/src/file.c
@@ -390,7 +390,7 @@ LoadPCB (char *Filename)
 
       /* clear 'changed flag' */
       SetChangedFlag (false);
-      PCB->Filename = strdup (Filename);
+      PCB->Filename = STRDUP (Filename);
       /* just in case a bad file saved file is loaded */
 
       units_mm = (PCB->Grid != (int) PCB->Grid) ? true : false;
diff --git a/src/misc.c b/src/misc.c
index 75fd377..2e471d0 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1920,7 +1920,7 @@ AttributePutToList (AttributeListType *list, const char *name, const char *value
 
   /* Now add the new attribute.  */
   i = list->Number;
-  list->List[i].name = strdup (name);
+  list->List[i].name = STRDUP (name);
   list->List[i].value = STRDUP (value);
   list->Number ++;
   return 0;
